### DISCUSSION TITLE
Use text, custom shapes or emojis as nodes

### DISCUSSION THREAD

**Discussion by Delapouite** [org-roam/org-roam-ui#96] [Feature request]
Hello

Let's say I have a fictional roam corpus, where I have gathered notes about vehicles.
I have tagged these notes with values like `Car`, `Bike`…

At first, in order to distinguish between these transportation types, I was using arbitrary colors.
Ok, so notes about Cars will be Red dots, notes about Boats will be Blue (because of the sea), notes about Planes will be … argggg another kind of Blue because of the sky…

This color palettes was not scaling well. So I though, what about display the tag name instead of the round disc symbolizing a node on the graph…

And finally, it strikes me : if I can replace a round disc by a dedicated text, what about using Emojis? It's even cooler and self explanatory!

🚗, 🚲, 🛫, ⛵…

I did hack a bit in order to have a quick proof of concept : 
![image](https://user-images.githubusercontent.com/39315/136422883-31af137a-e014-4483-bf8e-91fbd303b262.png)

This way I can clearly see in 1 sec relations between notes in a very satisfying manner.

The code is absolutely not ready yet, but would you accept a Pull Request that would enhance the current tagging system?
Right now we can associated a tag with a color. What about expanding on the idea by also associating a small text box where the user could type a short word, and therefore emojis to be displayed in place of nodes?

Thanks for your feedback on this proposal.

**Comment by tefkah** [comment]
Okay that's a really cool idea, I love it! That's wayyy more visual than even more colors, genius!

How's the performance of this? Not that it matters much as it would be a totally opt-in thing, but from what I've been able to tell is that doing more with the canvas = significant slowdowns, e.g. the labels.

But anyway, I love it and would totally accept such a pull request! I think it would be good to have three options for the emoji:
1. Replace the node with an emoji (i'm guessing you are overlaying the node, not replacing them? Not sure how you replace them and also have the labels work, unless you're completely rewriting it)
2. Add an emoji to the bottom right corner of the node
3. Add an emoji in front of the label

The latter two I could see myself use a lot more, since it's not as disruptive.

If you want help, to discuss and/or be more involved with the development of org-roam-ui you can join our telegram group! https://t.me/joinchat/RLxdY4LP5XNhZjgy

**Reply by tefkah** [reply]
We could let the user of course insert any kind of text, but I think it would be a good idea to have a dedicated emoji picker there and only make emoji possible to select, I think that makes for a better user experience.

**Comment by rickswe** [comment]
You can just add the emoji to the title of the node, is there something I'm missing here?

![image](https://i.imgur.com/md5ibId.png)
